unittests/DOSGuard.cpp

```diff
@@ -38,28 +39,26 @@ constexpr static auto JSONData = R"JSON(
             "sweep_interval": 1,
             "max_connections": 2,
             "max_requests": 3,
-            "whitelist": ["127.0.0.1"]
-        }
-    }
-)JSON";
-
-constexpr static auto JSONData2 = R"JSON(
-    {
-        "dos_guard": {
-            "max_fetches": 100,
-            "sweep_interval": 0.1,
-            "max_connections": 2,
-            "whitelist": ["127.0.0.1"]
+            "whitelist": [
+                "127.0.0.1"
+            ]
         }
     }
 )JSON";
 
 constexpr static auto IP = "127.0.0.2";
 
+struct MockWhitelistHandler
+{
+    MOCK_METHOD(bool, isWhiteListed, (std::string_view ip), (const));
+};
+
+using MockWhitelistHandlerType = NiceMock<MockWhitelistHandler>;
+
 class FakeSweepHandler
 {
 private:
-    using guard_type = BasicDOSGuard<FakeSweepHandler>;
+    using guard_type = BasicDOSGuard<MockWhitelistHandlerType, FakeSweepHandler>;
```

Comment:
This is not too important and i know i'm guilty of leaving a few of those in the codebase but consider renaming to `GuardType` instead of the snake case notation here 🥇